### ISSUE TITLE
how does the forward kinematics work?

### ISSUE THREAD

**Issue by LeLocTai** [pepijndevos/spacefox#1]
The fk code here is incredibly simple, which surprised me as I have researched this before and it usually is very complicated. I guess that is possible because the motion range in this application is small. Can you explain the method used, or give me a some links or keywords to learn more about it? Thanks!

**Comment by pepijndevos** [comment]
I took the engineering approach and linearized everything around the operating point haha

The code basically doesn't really do pose calculation, but rather just calculates some relative motions.
First I add up the potentiometers in 3 pairs of common mode (up/down) and differential mode (side to side) motion.
Then I just add up the components that contribute to any given axis.
So up/down is the sum of all common modes, and twisting is the sum of all differential modes.
And so forth for the other directions.
I don't even try very hard to match all the magnitudes, but I did kinda try to make sure that things kinda cancel out between axis.

So long story short, it's simple because it's wrong. But wrong in a practical way.
I guess it's not very exact if you try to combine various simultaneous motions.

I'd happily take a PR from someone who actually knows what they are doing. I briefly looked into forward kinematics and as you say, it looked basically impossible. Mathematically the knob could be upside down and that's a valid solution to the system of equations. Kind of logical, but kind of unexpected given how well documented and straightforward (if tedious) the reverse problem is.

**Comment by LeLocTai** [comment]
That's cool, thanks!
